app/api/helpers/custom_fields.py

```diff
@@ -245,3 +246,15 @@ def validate(self, value):
             self.validation_error = 'Value of %s is not in available choices'
             return False
         return True
+
+
+class Licence(CustomField):
+    """
+    Custom field that gets the licence details given the licence name
+    """
+    __schema_example__ = "{'name': 'Licence name' 'long_name': 'Long licence name', \
```

Comment:
I am not sure about using a dictionary as a custom field. Since you are base-classing CustomField, the `__schema_type__` is set to string. So you won't be able to pass something like `"license": {"f": "abc", "z": "aa"}` in API. Only something like `"license": "string_on_the_right_side"` will work.